spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientConfiguration.java

```diff
@@ -75,6 +75,9 @@
 	@Value("${ribbon.client.name}")
 	private String name = "client";
 
+	@Value("${ribbon.restclient.enabled:false}")
+	public static boolean RIBBON_RESTCLIENT_ENABLED;
```

Comment:
This seems odd to me that this is a static. Please remove static and just make it a normal field. In fact, it probably needs to live in a different place.

Reply:
Agreed, having this static is pretty strange, I guess the thing I struggle with is how to correctly inject this flag down into the command? 

The first version of this pull request, Pat had kind of flipped the problem around and just changed how "executeWithLoadBalancer()" behaves in the two retriable clients. That approach is nice because then we don't have to figure out how to inject that value.

We are totally willing to change it, just trying to figure out the best approach for injecting that flag down into the command.

Would you prefer we do something like: 

```java
public abstract class AbstractRibbonCommand<LBC extends AbstractLoadBalancerAwareClient<RQ, RS>, RQ extends ClientRequest, RS extends HttpResponse>
		extends HystrixCommand<ClientHttpResponse> implements RibbonCommand {
	protected final LBC client;
	protected RibbonCommandContext context;
	protected ZuulFallbackProvider zuulFallbackProvider;
	protected IClientConfig config;
	protected Boolean ribbonRestclientEnabled = false;
```

And then in the RestClientRibbonCommand: 

```java
public class RestClientRibbonCommand extends AbstractRibbonCommand<RestClient, HttpRequest, HttpResponse> {
	public RestClientRibbonCommand(String commandKey, RestClient client,
			RibbonCommandContext context, ZuulProperties zuulProperties) {
		super(commandKey, client, context, zuulProperties);
		ribbonRestclientEnabled = Boolean.TRUE;
	}
```

Reply:
I'd prefer something like a method in `AbstractRibbonCommand` like `boolean isExecuteWithLoadBalancer()` that by default returns false. `RestClientRibbonCommand` can then override.